multiqc/modules/vep/vep.py

```diff
@@ -0,0 +1,221 @@
+                    value = cells[1][4:-5]
+                    if key == "Novel / existing variants":
+                        values = value.split("/")
+                        novel = values[0].split("(")[0].replace(" ", "")
+                        existing = values[1].split("(")[0].replace(" ", "")
+                        self.vep_data[f["s_name"]][title]["Novel variants"] = int(novel)
+                        self.vep_data[f["s_name"]][title]["Existing variants"] = int(existing)
+                    else:
+                        self.vep_data[f["s_name"]][title][key] = int(value)
+
+    def parse_vep_txt(self, f):
+        """This Function will parse VEP summary files with plain text format"""
+        # Initialise vep_data dictionary for the current sample
+        if f["s_name"] not in self.vep_data:
+            self.vep_data[f["s_name"]] = {}
+        vepf = f["f"]
+        txt_data = {}
+        title = ""
+        for line in vepf.readlines():
+            line = line.rstrip("\n")
+            if len(line) < 1:
+                continue
+            if line[0] == "[":
+                title = line.replace("[", "").replace("]", "")
+                txt_data[title] = {}
+                continue
+            key, value = line.split("\t")
+            if key == "Novel / existing variants":
+                values = value.split("/")
+                novel = values[0].split("(")[0].replace(" ", "")
+                existing = values[1].split("(")[0].replace(" ", "")
+                txt_data[title]["Novel variants"] = int(novel)
+                txt_data[title]["Existing variants"] = int(existing)
+            elif title != "VEP run statistics":
+                txt_data[title][key] = int(value)
+            else:
+                txt_data[title][key] = value
+
+        for title in txt_data.keys():
+            if title not in self.vep_data[f["s_name"]]:
+                self.vep_data[f["s_name"]][title] = {}
+            self.vep_data[f["s_name"]][title] = txt_data[title]
+
+    def add_stats_table(self):
+        """Add a section with VEP General Statistics"""
+        table_config = {
+            "id": "vep-general-stats",  # ID used for the table
+            "table_title": "VEP General Statistics",  # Title of the table. Used in the column config modal
+            "save_file": False,  # Whether to save the table data to a file
+            "sortRows": True,  # Whether to sort rows alphabetically
+            "only_defined_headers": False,  # Only show columns that are defined in the headers config
+            "col1_header": "Sample Name",  # The header used for the first column
+            "no_beeswarm": True,  # Force a table to always be plotted (beeswarm by default if many rows)
+        }
+        title = "General statistics"
+        plot_cats = OrderedDict()
+        plot_data = OrderedDict()
+        color_list = ["Oranges", "Blues", "Reds", "Greens"]
+        for sample_name in self.vep_data:
```

Comment:
Yeah same here - this is _too_ efficient. A guideline for MultiQC colours is that they should match the type of data that they represent. So reds = things that are bad when there are lots. Greens = good, and so on.

Here you have meaningful colours assigned randomly. We also can't customise single columns, for example to hide specific ones and so on.